File: src/SmartSolar.Application/DTOs/Users/RegisterProsumerRequest.cs
/*
 * File: RegisterProsumerRequest.cs
 * Project: Smart Solar Microgrid Trading System
 * Purpose: Shared project source file for the SE4040 EAD implementation.
 * Note: Keep this header and add/update method-level comments as the code evolves.
 */

using System.ComponentModel.DataAnnotations;

namespace SmartSolar.Application.DTOs.Users;

public sealed class RegisterProsumerRequest
{
    [Required]
    [RegularExpression(@"^([0-9]{9}[VvXx]|[0-9]{12})$", ErrorMessage = "NIC must be a valid 9-digit legacy NIC with V/X or a 12-digit NIC.")]
    public string Nic { get; init; } = string.Empty;

    [Required, StringLength(120, MinimumLength = 2)]
    public string FullName { get; init; } = string.Empty;

    [Required, EmailAddress]
    public string Email { get; init; } = string.Empty;

    [Required, StringLength(20, MinimumLength = 7)]
    public string PhoneNumber { get; init; } = string.Empty;

    [Required, StringLength(100, MinimumLength = 8)]
    public string Password { get; init; } = string.Empty;
}

File: mobile/SmartSolarMobile/app/src/main/java/com/smartsolar/mobile/data/remote/dto/RegisterProsumerRequest.java
package com.smartsolar.mobile.data.remote.dto;

public final class RegisterProsumerRequest {
    private final String nic;
    private final String fullName;
    private final String email;
    private final String phoneNumber;
    private final String password;

    public RegisterProsumerRequest(String nic, String fullName, String email, String phoneNumber, String password) {
        this.nic = nic;
        this.fullName = fullName;
        this.email = email;
        this.phoneNumber = phoneNumber;
        this.password = password;
    }
}
